modules/multilanguage/src/main/scala/com/ukg/surge/multilanguage/MultilanguageGatewayServer.scala

```diff
@@ -7,39 +7,59 @@ import akka.event.{ Logging, LoggingAdapter }
 
-  def run(): Future[Http.ServerBinding] = {
-    implicit val sys: ActorSystem = system
-    implicit val ec: ExecutionContext = sys.dispatcher
+  implicit val sys: ActorSystem = system
+  implicit val ec: ExecutionContext = sys.dispatcher
 
-    val logger: LoggingAdapter = Logging(system, classOf[MultilanguageGatewayServer])
+  private val logger: LoggingAdapter = Logging(system, classOf[MultilanguageGatewayServer])
+  private val config = ConfigFactory.load()
 
-    val config = ConfigFactory.load()
-    val host = config.getString("surge-server.host")
-    val port = config.getInt("surge-server.port")
+  val aggregateName: String = config.getString("surge-server.aggregate-name")
+  val eventsTopicName: String = config.getString("surge-server.events-topic")
+  val stateTopicName: String = config.getString("surge-server.state-topic")
 
-    val aggregateName: String = config.getString("surge-server.aggregate-name")
-    val eventsTopicName: String = config.getString("surge-server.events-topic")
-    val stateTopicName: String = config.getString("surge-server.state-topic")
+  val businessLogicgRPCHost: String = config.getString("business-logic-server.host")
+  val businessLogicgRPCPort: Int = config.getInt("business-logic-server.port")
+  val businessLogicgRPCClientSettings: GrpcClientSettings = GrpcClientSettings.connectToServiceAt(businessLogicgRPCHost, businessLogicgRPCPort).withTls(false)
 
-    val businessLogicgRPCHost: String = config.getString("business-logic-server.host")
-    val businessLogicgRPCPort: Int = config.getInt("business-logic-server.port")
-    val businessLogicgRPCClientSettings: GrpcClientSettings = GrpcClientSettings.connectToServiceAt(businessLogicgRPCHost, businessLogicgRPCPort).withTls(false)
+  val bridgeToBusinessApp: BusinessLogicService = BusinessLogicServiceClient(businessLogicgRPCClientSettings)
+  val genericSurgeCommandBusinessLogic = new GenericSurgeCommandBusinessLogic(aggregateName, eventsTopicName, stateTopicName, bridgeToBusinessApp)
 
-    val bridgeToBusinessApp: BusinessLogicService = BusinessLogicServiceClient(businessLogicgRPCClientSettings)
+  lazy val surgeEngine: SurgeCommand[UUID, SurgeState, SurgeCmd, Nothing, SurgeEvent] = {
+    val engine = SurgeCommand(system, genericSurgeCommandBusinessLogic, system.settings.config)
+    logger.info("Started engine!")
+    engine.start()
+    engine
+  }
 
-    logger.info(
-      s"Binding multilanguage gateway server on $host:$port. Aggregate name: $aggregateName. Events topic: $eventsTopicName. State topic: $stateTopicName")
+  def doHealthCheck(): Future[HealthCheckReply] = {
+    for {
+      surgeHealth <- surgeEngine.healthCheck
+      businessAppHealth <- bridgeToBusinessApp.healthCheck(HealthCheckRequest()) if surgeHealth.isHealthy.getOrElse(false)
+      multilanguageServerHealth = businessAppHealth.copy(serviceName = "multilanguage-server") if businessAppHealth.status == Status.UP
+    } yield multilanguageServerHealth
```

Comment:
Hmm what happens on this if `surgeHealth` comes back unhealthy here? With the if's in here I'm wondering if we yield a `Future.successful(HealthCheckReply(status = DOWN))` or a `Future.failed(not sure what it'd fail with?)`